src/main/java/uk/gov/companieshouse/confirmationstatementapi/service/FilingService.java

```
@@ -0,0 +1,64 @@
+import java.util.Map;
+import java.util.Optional;
+
+@Service
+public class FilingService {
+
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy/MM/dd");
+    private ConfirmationStatementService confirmationStatementService;
+    private Environment environment;
+
+    @Autowired
+    public FilingService(ConfirmationStatementService csService, Environment env) {
+        confirmationStatementService = csService;
+        environment = env;
+    }
+
+    public FilingApi generateConfirmationFiling(String confirmationStatementId) throws SubmissionNotFoundException {
+        FilingApi filing = new FilingApi();
+        filing.setKind("Confirmation Statement");
+        setFilingApiData(filing, confirmationStatementId);
+        setDescription(filing);
+        return filing;
+    }
+
+    private void setFilingApiData(FilingApi filing, String confirmationStatementId) throws SubmissionNotFoundException {
+        Optional<ConfirmationStatementSubmissionJson> submissionOpt =
+                confirmationStatementService.getConfirmationStatement(confirmationStatementId);
+        if(submissionOpt.isEmpty()){
+            throw new SubmissionNotFoundException(
+                    String.format("Empty submission returned when generating filing for %s", confirmationStatementId));
+        }
+        ConfirmationStatementSubmissionJson submission = submissionOpt.get();
+        if(submission.getData() != null) {
+            Map<String, Object> data = new HashMap<>();
+            data.put("confirmationStatementDate", submission.getData().getMadeUpToDate());
+            data.put("tradingOnMarket", !submission.getData().getTradingStatusData().getTradingStatusAnswer());
+            data.put("dtr5Ind", false);
+            filing.setData(data);
+        }
+    }
+
+    private void setDescription(FilingApi filing) {
+        LocalDate madeUpToDate = (LocalDate)filing.getData().get("confirmationStatementDate");
```

setDescription runs after setFilingApiData, so filing.getData().get("confirmationStatementDate") can return null if it is not set in setFilingApiData and cause a null pointer exception. 

If submission.getData() == null in setFilingApiData we should probably throw an exception there as we won't be able to get the data to put in the filing.